### PR TITLE
fix(@desktop/wallet): solving the issues with SwapInputPanel list in the issue

### LINKED ISSUE #15162: Issues with SwapInputPanel
### Description

While integrating the SwapInputPanel Ive found the below issues -

- [x] Placeholder should be 0.00 for fiat and 0 for crypto value 
-> Already fixed under https://github.com/status-im/status-desktop/pull/15161
- [x] Need separate loading state for fiat value https://www.figma.com/design/TS0eQX9dAZXqZtELiwKIoK/Swap---Milestone-1?node-id=3407-245552&m=dev
- [x] When values entered is greater than balance then fiat value not calculated and that’s wrong https://www.figma.com/design/TS0eQX9dAZXqZtELiwKIoK/Swap---Milestone-1?node-id=4180-85777&m=dev
- [x] cryptoValue is also not reevaluated when amount entered > max Balance because of condition on line 105 in AmountToSend.qml
- [x] After values being set once and then setting tokenAmount to empty string leads to old string not being update 
https://github.com/status-im/status-desktop/assets/60327365/b3d77a77-db6c-4cce-a304-faa815dd450d
- [x] Incase max value is 0 it should be highlighted in red https://www.figma.com/design/TS0eQX9dAZXqZtELiwKIoK/Swap---Milestone-1?node-id=3430-338783&m=dev
- [x] consider renaming cryptoValue and cryptoValueRaw and cryptoValueValid as they are misleading should be generic as this could be fiat or crypto amounts
- [ ] Incase the user enters a small decimal value and resets the account selected, the text gets reset to 0

When 0.5 is entered its correct
https://github.com/status-im/status-desktop/assets/60327365/5d99a0c5-899b-48e2-9eb9-a26ded5bc366

When 0.005 is entered I think precision is lost or something
https://github.com/status-im/status-desktop/assets/60327365/6968148d-2b52-4f5c-af06-8797b0055af0

**NOTE** under this task please check all the TODO's blocked by this and enable in the UI https://github.com/status-im/status-desktop/issues/15162

### PR DESCRIPTION
**Pull request by Khushboo-dev-cpp**
fixes  #15162

### What does the PR do

Solves all the bugs mentioned under https://github.com/status-im/status-desktop/issues/15162

### Affected areas

SwapInputPanel.qml, SwapModal.qml

### StatusQ checklist

- [ ] add documentation if necessary (new component, new feature)
- [ ] update sandbox app
  - in case of new component, add new component page
  - in case of new features, add variation to existing component page
  - nice to have: add it to the demo application as well
- [ ] test changes in both light and dark theme?

### Screenshot of functionality (including design for comparison)

- [x] I've checked the design and this PR matches it

https://github.com/status-im/status-desktop/assets/60327365/12a06ab0-87f4-4c44-b07e-71c6279c167e

https://github.com/status-im/status-desktop/assets/60327365/2e4e87c0-a154-4818-b36a-6fe4924f1bcd

### CODE REVIEW

File: ui/imports/shared/controls/ErrorTag.qml [hunk 1]
```diff
@@ -13,6 +13,7 @@ InformationTag {
 
     property string text
     property string buttonText
+    property bool buttonVisible
```
Comment (caybro) [inline ui/imports/shared/controls/ErrorTag.qml:16]: Why you need an extra property here?

File: ui/app/AppLayouts/Wallet/popups/swap/SwapModal.qml [hunk 2]
```diff
@@ -34,15 +34,22 @@ StatusDialog {
     rightPadding: Style.current.xlPadding
     backgroundColor: Theme.palette.baseColor3
 
+    QtObject {
+        id: d
+        function fetchSuggestedRoutes() {
+            root.swapAdaptor.newFetchReset()
+            let correctlyFilled = root.swapInputParamsForm.formCorrectlyFilled && payPanel.valueValid
```
Comment (caybro) [inline ui/app/AppLayouts/Wallet/popups/swap/SwapModal.qml:41]:
```suggestion
            const correctlyFilled = root.swapInputParamsForm.formCorrectlyFilled && payPanel.valueValid
```

File: ui/app/AppLayouts/Wallet/popups/swap/SwapModal.qml [hunk 3]
```diff
@@ -34,15 +34,22 @@ StatusDialog {
...
+            Qt.callLater(() => root.swapAdaptor.fetchSuggestedRoutes(payPanel.rawValue, correctlyFilled))
```
Comment (caybro) [inline ui/app/AppLayouts/Wallet/popups/swap/SwapModal.qml:42]: Why `callLater()`?

File: ui/imports/shared/controls/ErrorTag.qml (hunk 1 above)
Comment (caybro) [reply ui/imports/shared/controls/ErrorTag.qml:16]: Ah I get it now :)

**Review by caybro — COMMENTED** [review]
Apart from this and some minor things looks good!

```
[2024-06-18T09:10:23.602Z] 3: FAIL!  : QmlTests::SwapModal::test_modal_pay_input_switching_accounts() Compared values are not the same
[2024-06-18T09:10:23.602Z] 3:    Actual   (): Buy crypto
[2024-06-18T09:10:23.602Z] 3:    Expected (): 
[2024-06-18T09:10:23.602Z] 3:    Loc: [/home/jenkins/workspace/s_linux_x86_64_tests-ui_PR-15227/storybook/qmlTests/tests/tst_SwapModal.qml(1129)]
```

Looks like a legit failure

File: ui/imports/shared/popups/send/views/AmountToSend.qml [hunk 4]
```diff
@@ -88,14 +89,11 @@ ColumnLayout {
...
         readonly property string cryptoValueRawToSend: {
-            if (!root.inputNumberValid)
-                return "0"
-
             return SQUtils.AmountsArithmetic.fromNumber(
                         d.cryptoValueToSend, root.multiplierIndex).toString()
         }
 
-        // Crypto value should be represented by 0 and fiat with 0.00
+	// Crypto value should be represented by 0 and fiat with 0.00
```
Comment (caybro) [inline ui/imports/shared/popups/send/views/AmountToSend.qml:96]: Weird indent

**Review by caybro — APPROVED** [review]
LGTM

**Review by dlipicar — APPROVED** [review]

**Pull request merged**